pkg/vsphere/vm/vm.go

```diff
@@ -116,3 +116,34 @@ func (vm *VirtualMachine) FetchExtraConfig(ctx context.Context) (map[string]stri
 	}
 	return info, nil
 }
+
+// WaitForExtraConfig waits until key shows up with the expected value inside the ExtraConfig
+func (vm *VirtualMachine) WaitForExtraConfig(ctx context.Context, key string, expected string) error {
```

Comment:
That's why we are passing context and it - to be able to define deadlines/timeouts or cancelation on top of it. We can key/expected string to a func that does the check so that we can do more enriched checks but not sure how it will help your original concern